californium-core/src/main/java/org/eclipse/californium/core/coap/OptionSet.java

```diff
@@ -1270,7 +1275,46 @@ public OptionSet removeObserve() {
 	public static boolean isValidObserveOption(final int value) {
 		return value >= 0 && value <= MAX_OBSERVE_NO;
 	}
-
+	
+	/**
+	 * Returns the byte array value of the OSCore option.
+	 * @return the OSCore value or null if the option is not present
+	 */
+	public byte[] getOscore() {
+		return oscore;
+	}
+	
+	/**
+	 * Checks if the OSCore option is present.
+	 * @return true if present
+	 */
+	public boolean hasOscore() {
+		return oscore != null;
+	}
+	
+	/**
+	 * Replaces the Oscore option with oscore.
+	 * Returns the current OptionSet object for a fluent API.
+	 * 
+	 * @param oscore the new Oscore value
+	 * @return this OptionSet
+	 */
+	public OptionSet setOscore(byte[] oscore){
+		if(oscore != null)
```

Comment:
The other setter are using exceptions to indicate argument errors. I would go for a NullPointerException. Please add that also to the javadoc.

Reply:
Any information on that?